From 0bf49c3dc00e07c05894b442c931fc3d40d806ed Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Fri, 24 Oct 2025 11:59:09 +0200
Subject: [PATCH] Introduce SamsungPropsHooks

UN1CA: this patch is not complete! Please check unica/mods/settings/customize.sh for more info.
---
 .../android/os/SystemProperties.smali         |  34 +-
 .../io/mesalabs/unica/SamsungPropsHooks.smali | 580 ++++++++++++++++++
 2 files changed, 608 insertions(+), 6 deletions(-)
 create mode 100644 smali_classes6/io/mesalabs/unica/SamsungPropsHooks.smali

diff --git a/smali_classes3/android/os/SystemProperties.smali b/smali_classes3/android/os/SystemProperties.smali
index 09f83663..7f1c0398 100644
--- a/smali_classes3/android/os/SystemProperties.smali
+++ b/smali_classes3/android/os/SystemProperties.smali
@@ -373,7 +373,7 @@
 .end method
 
 .method public static whitelist get(Ljava/lang/String;)Ljava/lang/String;
-    .locals 1
+    .locals 2
     .annotation runtime Landroid/annotation/SystemApi;
     .end annotation
 
@@ -381,18 +381,29 @@
 
     move-result-object v0
 
-    if-nez v0, :cond_0
+    invoke-static {p0}, Lio/mesalabs/unica/SamsungPropsHooks;->onSystemPropertiesGet(Ljava/lang/String;)Ljava/lang/String;
+
+    move-result-object v1
+
+    if-eqz v0, :cond_0
+
+    return-object v0
 
+    :cond_0
+    if-eqz v1, :cond_1
+
+    return-object v1
+
+    :cond_1
     invoke-static {p0}, Landroid/os/SystemProperties;->native_get(Ljava/lang/String;)Ljava/lang/String;
 
     move-result-object v0
 
-    :cond_0
     return-object v0
 .end method
 
 .method public static whitelist get(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
-    .locals 1
+    .locals 2
     .annotation runtime Landroid/annotation/SystemApi;
     .end annotation
 
@@ -400,13 +411,24 @@
 
     move-result-object v0
 
-    if-nez v0, :cond_0
+    invoke-static {p0}, Lio/mesalabs/unica/SamsungPropsHooks;->onSystemPropertiesGet(Ljava/lang/String;)Ljava/lang/String;
+
+    move-result-object v1
+
+    if-eqz v0, :cond_0
+
+    return-object v0
 
+    :cond_0
+    if-eqz v1, :cond_1
+
+    return-object v1
+
+    :cond_1
     invoke-static {p0, p1}, Landroid/os/SystemProperties;->native_get(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
 
     move-result-object v0
 
-    :cond_0
     return-object v0
 .end method
 
diff --git a/smali_classes6/io/mesalabs/unica/SamsungPropsHooks.smali b/smali_classes6/io/mesalabs/unica/SamsungPropsHooks.smali
new file mode 100644
index 00000000..a4dd3a99
--- /dev/null
+++ b/smali_classes6/io/mesalabs/unica/SamsungPropsHooks.smali
@@ -0,0 +1,580 @@
+.class public final Lio/mesalabs/unica/SamsungPropsHooks;
+.super Ljava/lang/Object;
+.source "SamsungPropsHooks.java"
+
+
+# static fields
+.field private static final blacklist AREMOJI_PACKAGE_NAME:Ljava/lang/String; = "com.samsung.android.aremoji"
+
+.field private static final blacklist CAMERAX_PACKAGE_NAME:Ljava/lang/String; = "com.samsung.android.cameraxservice"
+
+.field private static final blacklist CAMERA_PACKAGE_NAME:Ljava/lang/String; = "com.sec.android.app.camera"
+
+.field private static final blacklist DEBUG:Z = false
+
+.field private static final blacklist FM_RADIO_PACKAGE_NAME:Ljava/lang/String; = "com.sec.android.app.fm"
+
+.field private static final blacklist GPHOTOS_PACKAGE_NAME:Ljava/lang/String; = "com.google.android.apps.photos"
+
+.field private static final blacklist SINGLETAKE_PACKAGE_NAME:Ljava/lang/String; = "com.samsung.android.singletake.service"
+
+.field private static final blacklist SMART_VIEW_PACKAGE_NAME:Ljava/lang/String; = "com.samsung.android.smartmirroring"
+
+.field private static final blacklist TAG:Ljava/lang/String; = "SamsungPropsHooks"
+
+.field private static volatile blacklist sIsUserApp:Z
+
+.field private static volatile blacklist sPackageName:Ljava/lang/String;
+
+.field private static volatile blacklist sSpoofSingleTakeVars:Z
+
+
+# direct methods
+.method private constructor blacklist <init>()V
+    .locals 0
+
+    invoke-direct {p0}, Ljava/lang/Object;-><init>()V
+
+    return-void
+.end method
+
+.method private static blacklist dlog(Ljava/lang/String;)V
+    .locals 0
+
+    return-void
+.end method
+
+.method private static blacklist getDeviceModel()Ljava/lang/String;
+    .locals 2
+
+    const-string v0, "ro.product.odm.model"
+
+    invoke-static {v0}, Landroid/os/SemSystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
+
+    move-result-object v0
+
+    const-string v1, "ro.boot.em.model"
+
+    invoke-static {v1, v0}, Landroid/os/SemSystemProperties;->get(Ljava/lang/String;Ljava/lang/String;)Ljava/lang/String;
+
+    move-result-object v0
+
+    return-object v0
+.end method
+
+.method public static blacklist init(Landroid/content/Context;)V
+    .locals 5
+
+    invoke-virtual {p0}, Landroid/content/Context;->getPackageName()Ljava/lang/String;
+
+    move-result-object v0
+
+    invoke-static {v0}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
+
+    move-result v1
+
+    if-eqz v1, :cond_0
+
+    goto/16 :goto_2
+
+    :cond_0
+    sput-object v0, Lio/mesalabs/unica/SamsungPropsHooks;->sPackageName:Ljava/lang/String;
+
+    const/4 v1, 0x1
+
+    const/4 v2, 0x0
+
+    :try_start_0
+    invoke-virtual {p0}, Landroid/content/Context;->getPackageManager()Landroid/content/pm/PackageManager;
+
+    move-result-object p0
+
+    invoke-virtual {p0, v0, v2}, Landroid/content/pm/PackageManager;->getApplicationInfo(Ljava/lang/String;I)Landroid/content/pm/ApplicationInfo;
+
+    move-result-object p0
+
+    iget p0, p0, Landroid/content/pm/ApplicationInfo;->flags:I
+
+    and-int/lit16 p0, p0, 0x81
+
+    if-nez p0, :cond_1
+
+    move p0, v1
+
+    goto :goto_0
+
+    :cond_1
+    move p0, v2
+
+    :goto_0
+    sput-boolean p0, Lio/mesalabs/unica/SamsungPropsHooks;->sIsUserApp:Z
+    :try_end_0
+    .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
+
+    goto :goto_1
+
+    :catch_0
+    move-exception p0
+
+    new-instance v3, Ljava/lang/StringBuilder;
+
+    const-string v4, "sIsUserApp = false; "
+
+    invoke-direct {v3, v4}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-virtual {p0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
+
+    move-result-object p0
+
+    invoke-virtual {p0}, Ljava/lang/Class;->getName()Ljava/lang/String;
+
+    move-result-object p0
+
+    invoke-virtual {v3, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object p0
+
+    invoke-virtual {p0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object p0
+
+    invoke-static {p0}, Lio/mesalabs/unica/SamsungPropsHooks;->dlog(Ljava/lang/String;)V
+
+    sput-boolean v2, Lio/mesalabs/unica/SamsungPropsHooks;->sIsUserApp:Z
+
+    :goto_1
+    const-string p0, "com.samsung.android.singletake.service"
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    sput-boolean p0, Lio/mesalabs/unica/SamsungPropsHooks;->sSpoofSingleTakeVars:Z
+
+    const-string p0, "com.samsung.android.aremoji"
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    const-string v2, "MODEL"
+
+    if-eqz p0, :cond_2
+
+    invoke-static {}, Lio/mesalabs/unica/SamsungPropsHooks;->getDeviceModel()Ljava/lang/String;
+
+    move-result-object p0
+
+    invoke-static {v2, p0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    goto/16 :goto_2
+
+    :cond_2
+    const-string p0, "com.samsung.android.cameraxservice"
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_3
+
+    invoke-static {}, Lio/mesalabs/unica/SamsungPropsHooks;->getDeviceModel()Ljava/lang/String;
+
+    move-result-object p0
+
+    invoke-static {v2, p0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    goto/16 :goto_2
+
+    :cond_3
+    const-string p0, "com.sec.android.app.camera"
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_4
+
+    invoke-static {}, Lio/mesalabs/unica/SamsungPropsHooks;->getDeviceModel()Ljava/lang/String;
+
+    move-result-object p0
+
+    invoke-static {v2, p0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    const/16 p0, 0x23
+
+    invoke-static {p0}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
+
+    move-result-object p0
+	
+    const-string v0, "SEM_FIRST_SDK_INT"
+
+    invoke-static {v0, p0}, Lio/mesalabs/unica/SamsungPropsHooks;->setVersionField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    goto/16 :goto_2
+
+    :cond_4
+    const-string p0, "com.sec.android.app.fm"
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_5
+
+    const-string p0, "SM-A526B"
+
+    invoke-static {v2, p0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    goto/16 :goto_2
+
+    :cond_5
+    const-string p0, "com.google.android.apps.photos"
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_6
+
+    const-string p0, "persist.sys.unica.gphotos"
+
+    invoke-static {p0, v1}, Landroid/os/SemSystemProperties;->getBoolean(Ljava/lang/String;Z)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_6
+
+    const-string p0, "BRAND"
+
+    const-string v0, "google"
+
+    invoke-static {p0, v0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    const-string p0, "MANUFACTURER"
+
+    const-string v0, "Google"
+
+    invoke-static {p0, v0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    const-string p0, "DEVICE"
+
+    const-string v0, "marlin"
+
+    invoke-static {p0, v0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    const-string p0, "PRODUCT"
+
+    invoke-static {p0, v0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    const-string p0, "Pixel XL"
+
+    invoke-static {v2, p0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    const-string p0, "FINGERPRINT"
+
+    const-string v0, "google/marlin/marlin:10/QP1A.191005.007.A3/5972272:user/release-keys"
+
+    invoke-static {p0, v0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    goto :goto_2
+
+    :cond_6
+    sget-boolean p0, Lio/mesalabs/unica/SamsungPropsHooks;->sSpoofSingleTakeVars:Z
+
+    if-eqz p0, :cond_7
+
+    invoke-static {}, Lio/mesalabs/unica/SamsungPropsHooks;->getDeviceModel()Ljava/lang/String;
+
+    move-result-object p0
+
+    invoke-static {v2, p0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    goto :goto_2
+
+    :cond_7
+    const-string p0, "com.samsung.android.smartmirroring"
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_8
+
+    const-string p0, "TYPE"
+
+    const-string v0, "userdebug"
+
+    invoke-static {p0, v0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    goto :goto_2
+
+    :cond_8
+    sget-object p0, Landroid/os/Build;->MODEL:Ljava/lang/String;
+
+    const-string v1, "SM-S901"
+
+    invoke-virtual {p0, v1}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_9
+
+    sget-boolean p0, Lio/mesalabs/unica/SamsungPropsHooks;->sIsUserApp:Z
+
+    if-eqz p0, :cond_9
+
+    const-string p0, "com.sec"
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
+
+    move-result p0
+
+    if-nez p0, :cond_9
+
+    const-string p0, "com.samsung"
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->startsWith(Ljava/lang/String;)Z
+
+    move-result p0
+
+    if-nez p0, :cond_9
+
+    invoke-static {}, Lio/mesalabs/unica/SamsungPropsHooks;->getDeviceModel()Ljava/lang/String;
+
+    move-result-object p0
+
+    invoke-static {v2, p0}, Lio/mesalabs/unica/SamsungPropsHooks;->setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+
+    :cond_9
+    :goto_2
+    return-void
+.end method
+
+.method public static blacklist onSystemPropertiesGet(Ljava/lang/String;)Ljava/lang/String;
+    .locals 3
+
+    const/4 v0, 0x0
+
+    const-string v1, "ro.product.model"
+
+    invoke-virtual {p0, v1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result v1
+
+    if-eqz v1, :cond_0
+
+    sget-boolean v1, Lio/mesalabs/unica/SamsungPropsHooks;->sSpoofSingleTakeVars:Z
+
+    if-eqz v1, :cond_0
+
+    invoke-static {}, Lio/mesalabs/unica/SamsungPropsHooks;->getDeviceModel()Ljava/lang/String;
+
+    move-result-object v0
+
+    :cond_0
+    if-eqz v0, :cond_1
+
+    new-instance v1, Ljava/lang/StringBuilder;
+
+    invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
+
+    const-string v2, "Spoofing \""
+
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v1
+
+    invoke-virtual {v1, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v1
+
+    const-string v2, "\" prop to \""
+
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v1
+
+    invoke-virtual {v1, v0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v1
+
+    const-string v2, "\""
+
+    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v1
+
+    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object v1
+
+    invoke-static {v1}, Lio/mesalabs/unica/SamsungPropsHooks;->dlog(Ljava/lang/String;)V
+
+    :cond_1
+    return-object v0
+.end method
+
+.method private static blacklist setBuildField(Ljava/lang/String;Ljava/lang/Object;)V
+    .locals 2
+
+    new-instance v0, Ljava/lang/StringBuilder;
+
+    const-string v1, "Spoofing Build."
+
+    invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-virtual {v0, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v0
+
+    const-string v1, " to \""
+
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v0
+
+    invoke-virtual {v0, p1}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
+
+    move-result-object v0
+
+    const-string v1, "\""
+
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v0
+
+    invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object v0
+
+    invoke-static {v0}, Lio/mesalabs/unica/SamsungPropsHooks;->dlog(Ljava/lang/String;)V
+
+    :try_start_0
+    const-class v0, Landroid/os/Build;
+
+    invoke-virtual {v0, p0}, Ljava/lang/Class;->getDeclaredField(Ljava/lang/String;)Ljava/lang/reflect/Field;
+
+    move-result-object v0
+
+    const/4 v1, 0x1
+
+    invoke-virtual {v0, v1}, Ljava/lang/reflect/Field;->setAccessible(Z)V
+
+    const/4 v1, 0x0
+
+    invoke-virtual {v0, v1, p1}, Ljava/lang/reflect/Field;->set(Ljava/lang/Object;Ljava/lang/Object;)V
+
+    const/4 p1, 0x0
+
+    invoke-virtual {v0, p1}, Ljava/lang/reflect/Field;->setAccessible(Z)V
+    :try_end_0
+    .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
+
+    return-void
+
+    :catch_0
+    move-exception p1
+
+    new-instance v0, Ljava/lang/StringBuilder;
+
+    const-string v1, "Failed to spoof Build."
+
+    invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-virtual {v0, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object p0
+
+    invoke-virtual {p0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object p0
+
+    const-string v0, "SamsungPropsHooks"
+
+    invoke-static {v0, p0, p1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
+
+    return-void
+.end method
+
+.method private static blacklist setVersionField(Ljava/lang/String;Ljava/lang/Object;)V
+    .locals 2
+
+    new-instance v0, Ljava/lang/StringBuilder;
+
+    const-string v1, "Spoofing Build.VERSION."
+
+    invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-virtual {v0, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v0
+
+    const-string v1, " to \""
+
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v0
+
+    invoke-virtual {v0, p1}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
+
+    move-result-object v0
+
+    const-string v1, "\""
+
+    invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object v0
+
+    invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object v0
+
+    invoke-static {v0}, Lio/mesalabs/unica/SamsungPropsHooks;->dlog(Ljava/lang/String;)V
+
+    :try_start_0
+    const-class v0, Landroid/os/Build$VERSION;
+
+    invoke-virtual {v0, p0}, Ljava/lang/Class;->getDeclaredField(Ljava/lang/String;)Ljava/lang/reflect/Field;
+
+    move-result-object v0
+
+    const/4 v1, 0x1
+
+    invoke-virtual {v0, v1}, Ljava/lang/reflect/Field;->setAccessible(Z)V
+
+    const/4 v1, 0x0
+
+    invoke-virtual {v0, v1, p1}, Ljava/lang/reflect/Field;->set(Ljava/lang/Object;Ljava/lang/Object;)V
+
+    const/4 p1, 0x0
+
+    invoke-virtual {v0, p1}, Ljava/lang/reflect/Field;->setAccessible(Z)V
+    :try_end_0
+    .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
+
+    return-void
+
+    :catch_0
+    move-exception p1
+
+    new-instance v0, Ljava/lang/StringBuilder;
+
+    const-string v1, "Failed to spoof Build.VERSION."
+
+    invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
+
+    invoke-virtual {v0, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+
+    move-result-object p0
+
+    invoke-virtual {p0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+
+    move-result-object p0
+
+    const-string v0, "SamsungPropsHooks"
+
+    invoke-static {v0, p0, p1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Throwable;)I
+
+    return-void
+.end method
-- 
2.54.0

